### ISSUE TITLE
Force new user to change password at the first login

### ISSUE THREAD

**Issue by seek78** [dnnsoftware/Dnn.Platform#5520]
Hi,
it will be useful to have a setting that force new users to change password at first login.

**Comment by valadas** [comment]
This feature exists, after creating the user, you can go to the user setting and click on [Force Password Change] link.
![image](https://user-images.githubusercontent.com/6371568/216541453-e0de21d2-3969-4982-a4fe-32a9444057a5.png)

**Comment by seek78** [comment]
What I meant was for it to be automatic for newly created users, not manually as you indicated.

**Comment by valadas** [comment]
Oh, I misunderstood, reopening this for further discussion

**Comment by valadas** [comment]
If this gets done, it would be cool to also have an option to force password resets every x days...

**Comment by sleupold** [comment]
> If this gets done, it would be cool to also have an option to force password resets every x days...

frequent password changes are no longer suggested by security experts (and Microsoft)

**Comment by bdukes** [comment]
I agree with Sebastian that it's not a recommended configuration, but we do support that feature, @valadas (under _Security_, _Member Accounts_ tab, _Member Management_ sub-tab)
![Settings for with help bubble stating "Enter the number of days before a user must change their password. Enter 0 (zero) if the password should never expire."](https://user-images.githubusercontent.com/59507/216640582-d8734915-4e66-4b32-84cb-fbd3ebbfbb3f.png)

@seek78 are you thinking of a toggle when creating a user (just like you can select _Random Password_ you could select _Force Password Change_? Or just a higher-level policy that would always force the password change without a choice?

You could do this yourself by [implementing `IUserEventHandlers` to handle the `UserCreated` event](https://docs.dnncommunity.org/content/features/extensibility/event-system/index.html#user-events).

**Comment by valadas** [comment]
OMG, I actually used that feature in the past and totally forgot it existed...

**Comment by seek78** [comment]
> I agree with Sebastian that it's not a recommended configuration, but we do support that feature, @valadas (under _Security_, _Member Accounts_ tab, _Member Management_ sub-tab) ![Settings for with help bubble stating "Enter the number of days before a user must change their password. Enter 0 (zero) if the password should never expire."](https://user-images.githubusercontent.com/59507/216640582-d8734915-4e66-4b32-84cb-fbd3ebbfbb3f.png)
> 
> @seek78 are you thinking of a toggle when creating a user (just like you can select _Random Password_ you could select _Force Password Change_? Or just a higher-level policy that would always force the password change without a choice?
> 
> You could do this yourself by [implementing `IUserEventHandlers` to handle the `UserCreated` event](https://docs.dnncommunity.org/content/features/extensibility/event-system/index.html#user-events).

Hi,
the first one: a toggle that force the password change at first access of users.
Better if this setting is at portal level.

Thanks
